packages/keystone/src/fields/types/decimal/index.ts

```diff
@@ -132,6 +132,9 @@ export const decimal =
           if (val === null && validation?.isRequired) {
             args.addValidationError(`${fieldLabel} is required`);
           }
+          if (val === null && config.isNullable === false) {
+            args.addValidationError(`${fieldLabel} cannot be set to 'null'`);
```

Comment:
Can you make these messages do `${fieldLabel} is required` (as in part of the same isRequired check as above and similarly in the other fields with isNullable)